### PR TITLE
feat(wallet): favorites for saved addresses

### PR DESCRIPTION
**Pull request by stefandunca**
### Implements: #6546

Depends on `status-go`'s API change [in review PR #2844](https://github.com/status-im/status-go/pull/2844)

Notes

- Remove duplicate name instead of ESN
- Fixes elide address

### Affected areas

Wallet - Saved Addresses

### Screenshot of functionality (including design for comparison)

image

| Implementation | Design |
| :-: | :-: |
| image | image |

### CODE REVIEW

File: ui/imports/utils/Utils.qml [hunk 1]
```diff
@@ -652,6 +652,7 @@ QtObject {
     }
 
     function getElidedCompressedPk(publicKey) {
+        conse
```
Comment (caybro) [inline ui/imports/utils/Utils.qml:655]: Nope :)

**Review by caybro — CHANGES_REQUESTED** [review]
Looks good, just remove that typo :)

File: src/app_service/service/saved_address/service.nim [hunk 2]
```diff
@@ -56,13 +72,50 @@ proc createOrUpdateSavedAddress*(self: Service, name, address: string): string =
...
+    if not response.error.isNil:
+      raise newException(Exception, response.error.message)
+
+    self.updateAddresses()
+    return ""
+  except Exception as e:
+    let errDesription = e.msg
+    return errDesription
+
+proc setFavouriteAddressState*(self: Service, address: string, favourite: bool): string =
```
Comment (saledjenic) [inline src/app_service/service/saved_address/service.nim:88]: @stefandunca to me this name is somehow misleading... when you read it, it says `setFavouriteAddressState` and I expect that is should set something, but in the body I see that it add/delete address. Wdyt of renaming it to `proc addRemoveFavouriteAddress*(self: Service, address: string, add: bool)` or any other more intuitive option (an ideal case would be to have 2 procs, `addToFavourite(self: Service, address: string)` and `removeFromFavourite(self: Service, address: string)`?

Comment (stefandunca) [reply src/app_service/service/saved_address/service.nim]: You are right. It seems I moved the presentation layer abstraction to the service layer. Fixed it, please have another look

**Review by saledjenic — APPROVED** [review]

**Review by micieslak — APPROVED** [review]
LGTM. Tested, works fine! One minor comment left.

**Review by anastasiyaig — APPROVED** [review]

**Review by alexandraB99 — APPROVED** [review]

**Review by caybro — APPROVED** [review]
LGTM!

**Pull request merged**